File: app/src/main/java/com/example/reciperrecommender/pantry.java
package com.example.reciperrecommender;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.drawerlayout.widget.DrawerLayout;

import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;

import com.google.android.material.navigation.NavigationView;
import com.google.firebase.auth.FirebaseAuth;

public class pantry extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener{

    private ImageButton side_nave1;
    private DrawerLayout mDrawerLayout;
    private CardView search_recipes;
    private CardView recipe_list1;
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pantry);

        side_nave1 = (ImageButton) findViewById(R.id.side_nave);
        mDrawerLayout = (DrawerLayout) findViewById(R.id.pantry);
        search_recipes = (CardView) findViewById(R.id.search_recipes);
        recipe_list1 = (CardView) findViewById(R.id.see_list);

        mAuth = FirebaseAuth.getInstance();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);




        side_nave1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                mDrawerLayout.openDrawer(Gravity.LEFT);
            }
        });

        search_recipes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(pantry.this, recipe_display.class));
            }
        });


        recipe_list1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(pantry.this, recipe_list.class));
            }
        });
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {

        int id = item.getItemId();

        if (id == R.id.b1_pantry) {

            Intent intent = new Intent(this, pantry.class);
            startActivity(intent);

        } else if (id == R.id.b2_basket) {
            Intent intent = new Intent(this, Basket.class);
            startActivity(intent);

        } else if (id == R.id.b3_list) {
            Intent intent = new Intent(this, Shopping_List.class);
            startActivity(intent);

        } else if (id == R.id.b4_favourite) {
            Intent intent = new Intent(this, favourite_recipes.class);
            startActivity(intent);

        } else if (id == R.id.b5_settings) {
            Intent intent = new Intent(this, Settings.class);
            startActivity(intent);

        } else if (id == R.id.b6_logout) {
            FirebaseAuth.getInstance().signOut();
            Intent intent = new Intent(this, LoginActivity.class);
            startActivity(intent);
        }
        return true;

    }
}

File: app/src/main/java/com/aaditya/reciperecommender/pantry.kt
package com.aaditya.reciperecommender

import android.content.Intent
import android.os.Bundle
import android.view.Gravity
import android.view.MenuItem
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView
import com.google.firebase.auth.FirebaseAuth



class pantry : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    //private var auth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //auth = FirebaseAuth.getInstance();

        setContentView(R.layout.activity_pantry)




//        val open_side_nav = findViewById(R.id.side_nave) as ImageButton
//        open_side_nav.setOnClickListener{
//            val intent = Intent(this, side_nav::class.java)
//            startActivity(intent)
//        }

        val search_recipes = findViewById(R.id.search_recipes) as CardView

        search_recipes.setOnClickListener {
            val intent = Intent(this, recipe_list::class.java)
            startActivity(intent)

        }

        val see_list = findViewById(R.id.see_list) as CardView

        see_list.setOnClickListener {
            val intent = Intent(this, Basket::class.java)
            startActivity(intent)

        }

        val navigationView: NavigationView = findViewById(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(this)

        val side_btn = findViewById(R.id.side_nave) as ImageButton
        side_btn.setOnClickListener{
            val navDrawer: DrawerLayout = findViewById(R.id.pantry)
            navDrawer.openDrawer(Gravity.LEFT);
        }

    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            R.id.b1_pantry -> startActivity(Intent(this,pantry::class.java))
            R.id.b2_basket -> startActivity(Intent(this,Basket::class.java))
            R.id.b3_list -> startActivity(Intent(this,Shopping_List::class.java))
            R.id.b4_favourite -> startActivity(Intent(this,favourite_recipes::class.java))
            R.id.b5_settings -> startActivity(Intent(this,MainActivity::class.java))
            /*R.id.b6_logout -> {
                auth?.signOut();
                startActivity(Intent(this,LoginActivity::class.java))
            }*/
        }
//        drawer.closeDrawer(GravityCompat.START)
        return true


    }

}